app/src/main/java/org/greenstand/android/TreeTracker/database/TreeTrackerDAO.kt

```diff
@@ -33,9 +33,15 @@ interface TreeTrackerDAO {
     @Query("SELECT * FROM planter_info WHERE _id = :id")
     suspend fun getPlanterInfoById(id: Long): PlanterInfoEntity?
 
+    @Query("SELECT * FROM planter_info WHERE planter_identifier = :identity")
+    suspend fun getPlanterInfoById(identity: String): PlanterInfoEntity?
+
     @Query("SELECT * FROM planter_info WHERE power_user = 1")
     suspend fun getPowerUser(): PlanterInfoEntity?
 
+    @Query("SELECT COUNT(*) FROM planter_info WHERE planter_identifier = :identifier LIMIT 1")
+    suspend fun checkIfPlanterExists(identifier: String): Int
```

Comment:
We don't need this method. We can use the other method you created, and check if it returns null or not.